ginga/rv/plugins/WCSAxes.py

```diff
@@ -0,0 +1,305 @@
+        for name in self.linestyles:
+            combobox.append_text(name)
+        combobox.set_index(self.linestyles.index(self.linestyle))
+        combobox.add_callback('activated', self.set_linestyle_cb)
+
+        b.num_ra.set_text(str(self.num_ra))
+        b.num_ra.set_tooltip('Number of lines drawn for RA')
+        b.num_ra.add_callback('activated', lambda *args: self.set_num_ra())
+
+        b.num_dec.set_text(str(self.num_dec))
+        b.num_dec.set_tooltip('Number of lines drawn for DEC')
+        b.num_dec.add_callback('activated', lambda *args: self.set_num_dec())
+
+        fr.set_widget(w)
+        vbox.add_widget(fr, stretch=0)
+
+        fr = Widgets.Frame('Labels')
+        captions = (('Show label', 'checkbutton'),
+                    ('Font size:', 'label', 'Font size', 'entryset'),
+                    ('RA angle:', 'label', 'RA angle', 'entryset'),
+                    ('DEC angle:', 'label', 'DEC angle', 'entryset'))
+        w, b = Widgets.build_info(captions, orientation=orientation)
+        self.w.update(b)
+
+        b.show_label.set_state(self.show_label)
+        b.show_label.set_tooltip('Show/hide label')
+        b.show_label.add_callback('activated', self.toggle_label_cb)
+
+        b.font_size.set_text(str(self.fontsize))
+        b.font_size.set_tooltip('Labels font size')
+        b.font_size.add_callback(
+            'activated', lambda *args: self.set_fontsize())
+
+        b.ra_angle.set_text(str(self.ra_angle))
+        b.ra_angle.set_tooltip('Orientation in deg of RA labels')
+        b.ra_angle.add_callback('activated', lambda *args: self.set_ra_angle())
+
+        b.dec_angle.set_text(str(self.dec_angle))
+        b.dec_angle.set_tooltip('Orientation in deg of DEC labels')
+        b.dec_angle.add_callback(
+            'activated', lambda *args: self.set_dec_angle())
+
+        fr.set_widget(w)
+        vbox.add_widget(fr, stretch=0)
+
+        top.add_widget(sw, stretch=1)
+
+        btns = Widgets.HBox()
+        btns.set_spacing(3)
+
+        btn = Widgets.Button("Close")
+        btn.add_callback('activated', lambda w: self.close())
+        btns.add_widget(btn, stretch=0)
+        btn = Widgets.Button("Help")
+        btn.add_callback('activated', lambda w: self.help())
+        btns.add_widget(btn, stretch=0)
+        btns.add_widget(Widgets.Label(''), stretch=1)
+        top.add_widget(btns, stretch=0)
+
+        container.add_widget(top, stretch=1)
+
+        self.gui_up = True
+        self.redo()
+
+    def redo(self):
+        try:
+            obj = self.canvas.get_object_by_tag(self.overlaytag)
+        except Exception:
+            pass
+        else:
+            if obj.kind != 'wcsaxes':
+                return True
+        try:
+            self.canvas.delete_object_by_tag(self.overlaytag)
+        except Exception:
+            pass
+
+        # dc.WCSAxes will pick up image under the hood.
+        # UNTIL HERE - what about other params to adjust?
```

Comment:
@pllim, I would add the `WCSAxes` object right after you create the new canvas in the constructor.  No need to create a new one for every `redo()`.